modules/modules_test.go

```diff
@@ -63,16 +63,72 @@ func TestDependencies(t *testing.T) {
 	svc, err = mm.InitModuleServices("serviceA", "serviceB")
 	assert.Nil(t, err)
 	assert.Equal(t, 2, len(svc))
+	assert.Equal(t, []string{"serviceB"}, getStopDependenciesForModule("serviceA", svc))
+	assert.Equal(t, []string(nil), getStopDependenciesForModule("serviceB", svc))
 
 	svc, err = mm.InitModuleServices("serviceC")
 	assert.NoError(t, err)
 	assert.Equal(t, 3, len(svc))
+	assert.Equal(t, []string{"serviceB", "serviceC"}, getStopDependenciesForModule("serviceA", svc))
+	assert.Equal(t, []string{"serviceC"}, getStopDependenciesForModule("serviceB", svc))
+	assert.Equal(t, []string(nil), getStopDependenciesForModule("serviceC", svc))
 
 	// Test loading of the module second time - should produce the same set of services, but new instances.
 	svc2, err := mm.InitModuleServices("serviceC")
 	assert.NoError(t, err)
 	assert.Equal(t, 3, len(svc))
 	assert.NotEqual(t, svc, svc2)
+	assert.Equal(t, []string{"serviceB", "serviceC"}, getStopDependenciesForModule("serviceA", svc))
+	assert.Equal(t, []string{"serviceC"}, getStopDependenciesForModule("serviceB", svc))
+	assert.Equal(t, []string(nil), getStopDependenciesForModule("serviceC", svc))
+}
+
+func TestManaged_AddDependency_ShouldErrorOnCircularDependencies(t *testing.T) {
```

Comment:
would it be worth it to also test an indirect circular dependency?

For example:

Register the modules: serviceA, serviceB, serviceC
Register dependencies:
serviceA -> serviceB
serviceB -> serviceC
serviceC -> serviceA